src/Transport.ts

```diff
@@ -19,6 +19,24 @@ interface InternalTransportOptions extends TransportOptions
 	canProduceByKind: CanProduceByKind;
 }
 
+class ConsumerCreationTask
+{
+	consumerOptions: ConsumerOptions;
+	promise: Promise<Consumer>;
+	resolve?: (consumer: Consumer) => void;
+	reject?: (error: Error) => void;
```

Comment:
Shouldn't some of those be `private`?

Reply:
nop, we need to access all of them from `Transport`